src/frontend/src/app/pages/profile-edit/profile-edit.component.ts

```diff
@@ -10,73 +15,142 @@ import {
+            </div>
 
-      <div>
-        Social Media
-        <div formGroupName="socialMedia">
-          @for (key of socialMediaKeys; track key) {
-            <div>
-              <label [for]="key">{{ socialMediaInputs[key].label }}</label>
-              <input [id]="key" type="text" [formControlName]="key" />
+            <div class="flex flex-row mb-4 items-center">
+              <span class="w-1/3 h-6 font-bold">Neuron ID</span>
+              <span>{{ userProfile.neuronId }}</span>
+              <app-info-icon [infoText]="nonEditableInfo"></app-info-icon>
             </div>
           }
         </div>
-      </div>
 
-      <div>
-        <a title="Cancel your edits" routerLink="/"> Cancel </a>
-        <button type="submit">Save</button>
+        <form [formGroup]="profileForm" (ngSubmit)="onSubmit()">
+          <div class="flex flex-row">
+            <label for="username" class="w-1/3 font-bold">Username</label>
+
+            <div class="flex flex-col mb-3 w-2/3">
+              <input
+                id="username"
+                type="text"
+                formControlName="username"
+                class="mb-1"
+                [ngClass]="{
+                  'border-red-600 bg-red-100': isControlInvalid('username')
+                }"
+              />
+
+              <div class="h-4 text-red-600 text-xs">
+                @if (isControlInvalid('username')) {
+                  {{ getErrorMessage('username') }}
+                }
+              </div>
+            </div>
+          </div>
+
+          <div class="flex flex-row mb-4">
+            <label for="bio" class="w-1/3 font-bold">Bio</label>
+            <textarea
+              id="bio"
+              type="text"
+              formControlName="bio"
+              class="w-2/3 h-24 resize-y leading-24"
+            ></textarea>
+          </div>
+
+          <div class="py-5">
+            <h2
+              class="text-left text-lg sm:text-xl font-normal mb-4 text-gray-900"
+            >
+              Social Media
+            </h2>
+            <div formGroupName="socialMedia">
+              @for (key of socialMediaKeys; track key) {
+                <div class="flex mb-7 items-center">
+                  <label [for]="key" class="w-1/3 font-bold">{{
+                    socialMediaInputs[key].label
+                  }}</label>
+                  <input
+                    [id]="key"
+                    type="text"
+                    [formControlName]="key"
+                    class="w-2/3"
+                  />
+                </div>
+              }
+            </div>
+          </div>
+
+          <div class="flex items-center">
+            <a title="Cancel your edits" routerLink="/" class="ml-auto mr-4">
+              Cancel
+            </a>
+            <button
+              type="submit"
+              [title]="profileForm.valid ? 'Save' : 'Fix the validation errors'"
```

Comment:
Since the button already has `Save` as its content, I think it would be better to remove the redundant title when the form is valid.
```suggestion
              [title]="profileForm.invalid ? 'Fix the validation errors' : undefined"
```

Reply:
It's not for this PR, but I think it'll be better to just display a message above/below the button when the form is invalid.